tests/rule_based_profiler/data_assistant/test_volume_data_assistant.py

```diff
@@ -363,7 +2578,7 @@ def test_execution_time_within_proper_bounds(
     data_assistant_result: DataAssistantResult = data_assistant.run()
 
     assert (
-        0.0 < data_assistant_result.execution_time <= 2.0
+        0.0 < data_assistant_result.execution_time <= 7.0
```

Comment:
Oof did we lose that much performance? I'd like @talagluck's input to see what an appropriate performance benchmark is.

Reply:
@cdkini Several things -- First, some kind of a caching solution (BatchMetricStore and/or BatchManager) will enable us to compute it once and reuse multiple times, until recomputing is deemed required.  Second, computing the entire profiler every time, sequentially, is a shortcoming we will have to address at some point.

Reply:
@cdkini Per @NathanFarmer -- going to use `@pytest.mark.timeout(300)` for that test to prevent failures, no matter how long the test could take.  Thanks.